src/databricks/sql/thrift_backend.py

```diff
@@ -1046,7 +1046,7 @@ def fetch_results(
 
         resp = self.make_request(self._client.FetchResults, req)
         if resp.results.startRowOffset > expected_row_start_offset:
-            logger.warning(
+            raise ValueError(
                 "Expected results to start from {} but they instead start at {}".format(
```

Comment:
can you also add the code to stop retry on fetch request next when it is inline mode

Reply:
as FetchResultNext is not idempotent, so you should not retry regardless of the status code.